src/Components/Server/src/Endpoints/CircuitMatcherPolicy.cs

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
```

Comment:
Would definitely like to see @rynowak's review on all the endpoint-related stuff, as it's quite far outside my area of expertise.